tools/kagami/src/main.rs

```diff
@@ -0,0 +1,298 @@
+                .multiple(false)
+        ).get_matches();
+    app
+}
+
+mod crypto {
+
+    use color_eyre::eyre::{eyre, WrapErr as _};
+    use iroha_crypto::{Algorithm, KeyGenConfiguration, KeyPair, PrivateKey};
+
+    pub fn key_pair(matches: &clap::ArgMatches) -> color_eyre::Result<KeyPair> {
+        let seed_option = matches.value_of("seed");
+        let private_key_option = matches.value_of("private-key");
+        let algorithm = matches
+            .value_of("algorithm")
+            .ok_or_else(|| eyre!("Failed to get algorithm name."))?
+            .parse::<Algorithm>()
+            .wrap_err("Failed to parse algorithm.")?;
+        let key_gen_configuration = KeyGenConfiguration::default().with_algorithm(algorithm);
+        let keypair: KeyPair = seed_option.map_or_else(
+            || -> color_eyre::Result<_> {
+                private_key_option.map_or_else(
+                    || {
+                        KeyPair::generate_with_configuration(key_gen_configuration.clone())
+                            .wrap_err("failed to generate key pair")
+                    },
+                    |private_key| {
+                        KeyPair::generate_with_configuration(
+                            key_gen_configuration
+                                .clone()
+                                .use_private_key(PrivateKey::from_hex(
+                                    algorithm,
+                                    &hex::decode(private_key)
+                                        .wrap_err("Failed to decode private key")?,
+                                )?),
+                        )
+                        .wrap_err("Failed to generate key pair")
+                    },
+                )
+            },
+            |seed| -> color_eyre::Result<_> {
+                KeyPair::generate_with_configuration(
+                    key_gen_configuration
+                        .clone()
+                        .use_seed(seed.as_bytes().into()),
+                )
+                .wrap_err("Failed to generate key pair")
+            },
+        )?;
+        Ok(keypair)
+    }
+}
+
+mod genesis {
+    use iroha_core::{
+        genesis::{RawGenesisBlock, RawGenesisBlockBuilder},
+        tx::{AssetDefinition, MintBox},
+    };
+
+    pub fn generate_default() -> color_eyre::Result<RawGenesisBlock> {
+        let asset_definition = AssetDefinition::quantity("rose#wonderland".parse()?).build();
+        let mut result = RawGenesisBlockBuilder::new()
+            .domain("wonderland".parse()?)
+            .with_account("alice".parse()?, crate::DEFAULT_PUBLIC_KEY.parse()?)
+            .with_asset(asset_definition.clone())
+            .finish_domain()
+            .build();
+        let mint = MintBox::new(
+            iroha_data_model::prelude::Value::U32(13_u32),
+            iroha_data_model::IdBox::AssetId(iroha_data_model::prelude::AssetId::new(
+                asset_definition.id().clone(), // Probably redundant clone
```

Comment:
Yep, seems to be redundant. Can't see a reason why compiler/clippy likes to leave cloning

Reply:
it's not redundant, it's because `id()` returns a reference

Reply:
The best solution is to make `id()` return a wrapper around an interned string (which is `Copy`).